includes/wikia/nirvana/WikiaView.class.php

```diff
@@ -114,36 +109,106 @@ protected function buildTemplatePath( $controllerClass, $methodName, $forceRebui
 			}
 
 			// Service classes must be dispatched by full name otherwise we default to a controller.
-			$controllerBaseName = $app->getBaseName( $controllerClass );
-			if ($app->isService($controllerClass)) {
-				$controllerClass = $app->getServiceClassName( $controllerBaseName );
-			} else {
-				$controllerClass = $app->getControllerClassName( $controllerBaseName );
+			$controllerClass = $this->normalizeControllerClass( $controllerClass );
+
+			$templateExists = false;
+			$templatePath = '';
+			$templates = $this->getTemplateOptions( $controllerClass, $methodName );
+			$dirName = $this->getTemplateDir( $controllerClass );
+			foreach ( $templates as $templateName ) {
+				$templatePath = $dirName . '/' . $templateName . '.' . $extension;
+
+				if ( file_exists( $templatePath ) ) {
+					$templateExists = true;
+					break;
+				}
 			}
 
-			if( empty( $wgAutoloadClasses[$controllerClass] ) ) {
-				throw new WikiaException( "Invalid controller or service name: {$controllerClass}" );
+			if ( !$templateExists ) {
+				throw new WikiaException( "Template file not found: {$templatePath}" );
 			}
 
-			// First we look for BaseName_MethodName
-			$dirName = dirname( $wgAutoloadClasses[$controllerClass] );
-			$basePath = "{$dirName}/templates/{$controllerBaseName}_{$methodName}";
-			$templatePath = "{$basePath}.$extension";
-			$templateExists = file_exists( $templatePath );
+			$this->setTemplatePath( $templatePath );
+		}
+		wfProfileOut(__METHOD__);
```

Comment:
Given this is a pretty core function of our Nirvana framework and Macbre says he still uses this for somethings, I'm hesitant to remove this.